x/ibc-rate-limit/contracts/rate-limiter/src/contract.rs

```diff
@@ -0,0 +1,527 @@
+        let res = instantiate(deps.as_mut(), mock_env(), info, msg).unwrap();
+        assert_eq!(0, res.messages.len());
+
+        // The ibc and gov modules are properly stored
+        assert_eq!(IBCMODULE.load(deps.as_ref().storage).unwrap(), IBC_ADDR);
+        assert_eq!(GOVMODULE.load(deps.as_ref().storage).unwrap(), GOV_ADDR);
+    }
+
+    #[test]
+    fn permissions() {
+        let mut deps = mock_dependencies();
+
+        let quota = QuotaMsg::new("Weekly", RESET_TIME_WEEKLY, 10, 10);
+        let msg = InstantiateMsg {
+            gov_module: Addr::unchecked(GOV_ADDR),
+            ibc_module: Addr::unchecked(IBC_ADDR),
+            channels: vec![Channel {
+                name: "channel".to_string(),
+                quotas: vec![quota],
+            }],
+        };
+        let info = mock_info(IBC_ADDR, &vec![]);
+        instantiate(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+
+        let msg = ExecuteMsg::SendPacket {
+            channel_id: "channel".to_string(),
+            channel_value: 3_000,
+            funds: 300,
+        };
+
+        // This succeeds
+        execute(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+
+        let info = mock_info("SomeoneElse", &vec![]);
+
+        let msg = ExecuteMsg::SendPacket {
+            channel_id: "channel".to_string(),
+            channel_value: 3_000,
+            funds: 300,
+        };
+        let err = execute(deps.as_mut(), mock_env(), info.clone(), msg).unwrap_err();
+        assert!(matches!(err, ContractError::Unauthorized { .. }));
+    }
+
+    #[test]
+    fn consume_allowance() {
+        let mut deps = mock_dependencies();
+
+        let quota = QuotaMsg::new("weekly", RESET_TIME_WEEKLY, 10, 10);
+        let msg = InstantiateMsg {
+            gov_module: Addr::unchecked(GOV_ADDR),
+            ibc_module: Addr::unchecked(IBC_ADDR),
+            channels: vec![Channel {
+                name: "channel".to_string(),
+                quotas: vec![quota],
+            }],
+        };
+        let info = mock_info(GOV_ADDR, &vec![]);
+        let _res = instantiate(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+
+        let msg = ExecuteMsg::SendPacket {
+            channel_id: "channel".to_string(),
+            channel_value: 3_000,
+            funds: 300,
+        };
+        let info = mock_info(IBC_ADDR, &vec![]);
+        let res = execute(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+        let Attribute { key, value } = &res.attributes[2];
+        assert_eq!(key, "weekly_used");
+        assert_eq!(value, "300");
+
+        let msg = ExecuteMsg::SendPacket {
+            channel_id: "channel".to_string(),
+            channel_value: 3_000,
+            funds: 300,
+        };
+        let err = execute(deps.as_mut(), mock_env(), info, msg).unwrap_err();
+        assert!(matches!(err, ContractError::RateLimitExceded { .. }));
+    }
+
+    #[test]
+    fn symetric_flows_dont_consume_allowance() {
```

Comment:
nit: Can we have one liner comments for some of the test cases on what we're trying to test?

Reply:
added